### PR TITLE
Docs for implements

### PR DESCRIPTION
**Pull request by czoido**
Docs for https://github.com/conan-io/conan/pull/14320

### CODE REVIEW

File: reference/conanfile/methods/package_id.rst [hunk 1]
```diff
@@ -12,14 +12,40 @@ The general rule is that every different value of ``settings`` and ``options`` c
...
+Then, if no ``package_id()`` method is specified in the recipe, Conan will
+automatically manage the fPIC setting in the ``package_id`` step like this:
+
+.. code-block:: python
+
+    if conanfile.options.get_safe("header_only") or conanfile.package_type is PackageType.HEADER:
+        conanfile.info.clear()
+
 
 To opt-out from this behavior, the method can be empty-defined:
```
Comment (memsharded) [inline reference/conanfile/methods/package_id.rst:50]: Remove this? No longer need to opt-out

File: reference/conanfile/methods/configure.rst [hunk 2]
```diff
@@ -43,35 +43,70 @@ so it should be removed:
...
+    if conanfile.options.get_safe("header_only"):
+        conanfile.options.rm_safe("fPIC")
+        conanfile.options.rm_safe("shared")
+    elif conanfile.options.get_safe("shared"):
+        conanfile.options.rm_safe("fPIC")
+
+Be aware that adding this implementation to the recipe may also affect the
+:ref:`configure<reference_conanfile_methods_config_options_implementations>` step.
+
 To opt-out from this behavior, the method can be empty-defined:
```
Comment (memsharded) [inline reference/conanfile/methods/configure.rst:90]: Remove, no need to opt-out now

File: reference/conanfile/methods/config_options.rst [hunk 3]
```diff
@@ -18,33 +18,62 @@ The ``config_options()`` method executes:
...
+
+.. code-block:: python
+
+    if conanfile.settings.get_safe("os") == "Windows":
+        conanfile.options.rm_safe("fPIC")
+
+Be aware that adding this implementation to the recipe may also affect the
+:ref:`configure<reference_conanfile_methods_configure_implementations>` step.
+
 To opt-out from this behavior, the method can be empty-defined:
```
Comment (memsharded) [inline reference/conanfile/methods/config_options.rst:60]: Remove, no longer need to opt-out

File: reference/conanfile/methods/package_id.rst [hunk 4]
```diff
@@ -12,30 +12,50 @@ The general rule is that every different value of ``settings`` and ``options`` c
...
+    if conanfile.options.get_safe("header_only") or conanfile.package_type is PackageType.HEADER:
+        conanfile.info.clear()
 
-To manage the package ID info further, the tool has to be explicitly called:
+If you need to implement custom behaviors in your recipes but also need this logic, it
+must be explicitly declared:
 
 .. code-block:: python
 
     from conan.tools import default_package_id
```
Comment (czoido) [inline reference/conanfile/methods/package_id.rst:54]:
```suggestion
```

**Review by memsharded — APPROVED** [review]

**Pull request merged**